tools/perf/dashboard_auth.py

```diff
@@ -0,0 +1,165 @@
+#   version: "version:1.2.0"
+# >
+#
+# wheel: <
+#   name: "infra/python/wheels/oauthlib-py2_py3"
+#   version: "version:3.0.1"
+# >
+#
+# wheel: <
+#   name: "infra/python/wheels/google-auth-httplib2-py2_py3"
+#   version: "version:0.0.3"
+# >
+# [VPYTHON:END]
+#
+# Copyright (c) 2022 The Brave Authors. All rights reserved.
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# you can obtain one at http://mozilla.org/MPL/2.0/.
+"""A tool to perform authorization to https://brave-perf-dashboard.appspot.com/
+
+The tool performs OAuth2 flow, stores/loads credentials, refreshes a token.
+Returns a valid token to stdout if succeeded.
+"""
+
+import os
+import sys
+import json
+
+# pylint: disable=no-name-in-module,import-error
+from components import path_util
+with path_util.SysPath(path_util.GetGoogleAuthDir(), 0):
+  from google.auth.transport.requests import Request
+  from google.oauth2.credentials import Credentials
+  from google_auth_oauthlib.flow import InstalledAppFlow
+# pylint: enable=no-name-in-module,import-error
+
+SCOPES = ['openid', 'https://www.googleapis.com/auth/userinfo.email']
+CLIENT_ID_FILE = os.path.join(os.path.expanduser("~"),
+                              '.perf_dashboard_client_id.json')
+PERF_CREDENTIAL_FILE = os.path.join(os.path.expanduser("~"),
+                                    '.perf_dashboard_credentials.json')
+
+
+def to_json(credentials):
+  prep = {
+      "token": credentials.token,
+      "refresh_token": credentials.refresh_token,
+      "token_uri": credentials.token_uri,
+      "client_id": credentials.client_id,
+      "client_secret": credentials.client_secret,
+      "scopes": credentials.scopes,
+  }
+  if credentials.expiry:  # flatten expiry timestamp
+    prep["expiry"] = credentials.expiry.isoformat() + "Z"
+
+  # Remove empty entries (those which are None)
+  prep = {k: v for k, v in prep.items() if v is not None}
+
+  return json.dumps(prep)
+
+
+def _GetDashboardCredentials(can_be_interactive=False):
+  credentials = None
+  should_store_new_credentials = False
+
+  if os.path.exists(PERF_CREDENTIAL_FILE):
+    credentials = Credentials.from_authorized_user_file(PERF_CREDENTIAL_FILE,
+                                                        SCOPES)
+
+  if credentials and not credentials.valid and credentials.refresh_token:
+    credentials.refresh(Request())
+    should_store_new_credentials = True
+
+  if not credentials or not credentials.valid or credentials.expired:
+    if can_be_interactive:
+      flow = InstalledAppFlow.from_client_secrets_file(CLIENT_ID_FILE, SCOPES)
+      credentials = flow.run_console()
+      should_store_new_credentials = True
+    else:
+      raise RuntimeError('No valid credentials for brave-perf-dashboard')
+
+  if should_store_new_credentials:
+    with open(PERF_CREDENTIAL_FILE, 'w') as credentials_dat:
```

Comment:
where is this running? is there no way to store these encrypted?

Reply:
It suppose to run on perf bots (currently it's my table slow laptop).
In future it could be another physical devices or Amazon EC2 instances.

 The current solution is temporary and uses authorization from a user.
 For bots a service account will be used instead: https://developers.google.com/identity/protocols/oauth2/service-account

Reply:
@atuchin-m who is managing the service account? your personal brave.com address or something else?

Reply:
@diracdeltas 
I believe currently nobody expect devops team has access to it. It's Google Cloud project named brave-perf-dashboard. Chris Hudson created a service account, but we haven't tested it on  a bot.

Could you suggest how service account credentials should be stored and passed to script in case of a physical PC?

Reply:
Maybe better to change the PR to remove `tools/perf/dashboard_auth.py` here and make another PR to keep thing more clear?
Also I could change the code to use a service account instead of personal.

Reply:
sgtm, also i'll start a discussion about the devops side on slack so it's not in a public thread